### PR TITLE
Support K8s 1.25

### PR DESCRIPTION
**Pull request by conradhanson**
Istio version 1.15 started to use pointer slices for the Items field in its api, which was incompatible with current templates. 

Changes include:
- updated go templates to support possibility of an external api having pointer slices for Items field
- go dependencies updated to k8s 1.25
- istio dependencies updated to 1.15.4

### CODE REVIEW

File: .github/workflows/pull_request.yaml [hunk 1]
```diff
@@ -11,10 +11,10 @@ jobs:
     name: Generate Code
     runs-on: ubuntu-18.04
     steps:
-    - name: Set up Go 1.16
+    - name: Set up Go 1.18
```
Comment (fernandosanchezs) [inline .github/workflows/pull_request.yaml:14]: is Go 1.18 intentional here instead of 1.19.4?

Comment (conradhanson) [reply .github/workflows/pull_request.yaml:14]: go mod is only at 18, updated to 18 before I found the need in gme to bump to 19. I can see if 19 is fine here and bump if so, just for dev simplicity b/t repos.

**Review by EItanya — COMMENTED** [review]
Makes sense to me, letting @ilackarms take a look as well

File: .github/workflows/pull_request.yaml (hunk 1 above)
Comment (conradhanson) [reply .github/workflows/pull_request.yaml:14]: bumped to 1.19. nothing changes / is affected so that is nice.

**Review by ilackarms — APPROVED** [review]

**Pull request merged**